openedx/core/djangoapps/course_views/course_views.py

```diff
@@ -0,0 +1,176 @@
+"""
+Tabs for courseware.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+from xmodule.tabs import CourseTab, link_reverse_func
+
+_ = lambda text: text
+
+
+# Stevedore extension point namespaces
+COURSE_VIEW_TYPE_NAMESPACE = 'openedx.course_view_type'
+
+
+class CourseViewType(object):
+    """
+    Base class of all course view type plugins.
```

Comment:
The big downside of Descriptor is that it's a) used by XModule, and b) used by Python. The existing overlap can be confusing already.

If we're going with calling these Tabs, then maybe we can merge the "Descriptor" and the "Tab" (and just call them both "Tab"), and then just require them to expose their data in such a way that mobile can consume them to construct whatever the UX metaphor on mobile is. That way we don't need two names for the python objects.